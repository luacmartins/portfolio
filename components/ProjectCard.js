const ProjectCard = ({ data }) => {
   return (
      <div>
         <div>
            <header className="font-semibold text-2xl leading-tight mt-2 sm:mt-0">{data.title}</header>
            <p className="mt-2 font-thin text-lg">{data.text}</p>
         </div>
         <div className="flex justify-start mt-2 py-2 sm:py-3">
            <a className="bg-blue-500 rounded px-2 py-1 text-white sm:w-32 sm:text-center hover:shadow hover:bg-blue-600" href={data.demo} alt={data.demoAlt} target="_blank">Live demo</a>
            <a className="border border-blue-500 px-2 py-1 text-blue-500 rounded ml-4 sm:w-32 sm:text-center hover:shadow hover:text-blue-600 hover:border-blue-600" href={data.code} alt={data.codeAlt} target="_blank">Source code</a>
         </div>
      </div>
   );
}

export default ProjectCard;