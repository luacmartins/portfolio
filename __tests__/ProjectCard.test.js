import ProjectCard from '../components/ProjectCard'
import renderer from 'react-test-renderer'

const data = {
   title: 'CM Photo',
   date: '2020-02-10',
   demo: 'website',
   demoAlt: 'live demo',
   code: 'github',
   codeAlt: 'source code',
   text: "I designed and wrote the code for my online photography store. I used React's Next.js framework and TailwindCSS to stactically generate the front end. State is managed with React hooks and data with markdown files. Testing is done using Jest. Fulfillment and payment were integrated using Rest APIs."
}

test('should render card correctly', () => {
   const component = renderer.create(<ProjectCard data={data} />)
   expect(component).toMatchSnapshot()
}) 