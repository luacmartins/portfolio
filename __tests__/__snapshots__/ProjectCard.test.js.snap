// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card correctly 1`] = `
<div>
  <div>
    <header
      className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
    >
      CM Photo
    </header>
    <p
      className="mt-2 font-thin text-lg"
    >
      I designed and wrote the code for my online photography store. I used React's Next.js framework and TailwindCSS to stactically generate the front end. State is managed with React hooks and data with markdown files. Testing is done using Jest. Fulfillment and payment were integrated using Rest APIs.
    </p>
  </div>
  <div
    className="flex justify-start mt-2 py-2 sm:py-3"
  >
    <a
      alt="live demo"
      className="bg-blue-500 rounded px-2 py-1 text-white sm:w-32 sm:text-center hover:shadow hover:bg-blue-600"
      href="website"
      target="_blank"
    >
      Live demo
    </a>
    <a
      alt="source code"
      className="border border-blue-500 px-2 py-1 text-blue-500 rounded ml-4 sm:w-32 sm:text-center hover:shadow hover:text-blue-600 hover:border-blue-600"
      href="github"
      target="_blank"
    >
      Source code
    </a>
  </div>
</div>
`;
